pkg/controller/certificates/sync.go

```diff
@@ -104,6 +107,30 @@ func (c *Controller) Sync(ctx context.Context, crt *v1alpha1.Certificate) (err e
 		return nil
 	}
 
+	// check that certificate secret name is unique
+	namespaceCrts := c.certificateLister.Certificates(crtCopy.Namespace)
+	otherCrts, err := namespaceCrts.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	hasDuplicate := false
+	for _, otherCrt := range otherCrts {
+		if otherCrt.Name != crtCopy.Name && otherCrt.Spec.SecretName == crtCopy.Spec.SecretName {
```

Comment:
It'd be awesome if we can also include *what* the other secret name is in at least the Event. I think including the name of the other certificate in the *status* could cause flapping in future (i.e. we could keep updating with different secret names each time) so may be best to avoid it there.